### PR TITLE
Data grid columns demos rewrite

### PR DESCRIPTION
**Pull request by pavel-2020**

### CODE REVIEW

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
 
-The customizations mentioned above are implemented in&nbsp;the <a href="/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/">editing</a> object for standard <b>Edit</b> and <b>Delete</b> buttons, and in&nbsp;the <a href="/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/buttons/">buttons</a> array for the custom <b>Clone</b> button.
\ No newline at end of file
+**Predefined buttons**
+The predefined buttons are edit and delete. You can enable them for all rows. For this, set the **editing**.[allowUpdating](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowUpdating) and **editing**.[allowDeleting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowDeleting) options to **true**. If you need to hide any of them for certain rows, implement a function with custom logic and assign it to the desired option. In this demo, such a function is assigned to the **allowDeleting** option.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md]:
```suggestion
The predefined buttons are Edit and Delete. You can enable them for all rows. For this, set the **editing**.[allowUpdating](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowUpdating) and **editing**.[allowDeleting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowDeleting) options to **true**. If you need to hide any of them for certain rows, implement a function with custom logic and assign it to the desired option. In this demo, such a function is assigned to the **allowDeleting** option.
```

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+
+The predefined editing buttons are displayed as textual links by default. To use icons instead, set **editing**.[useIcons](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#useIcons) to **true**.
+
+**Custom buttons**
+If the edit column should contain custom buttons, add its configuration to the [columns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/) array. DataGrid will display it according to its position in this array. Specify the column's [type](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#type) as *"buttons"* and declare the [buttons](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/buttons/) array. It should contain your custom buttons and values for the predefined buttons.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md]:
```suggestion
If the edit column should contain custom buttons, add its configuration to the [columns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/) array. **DataGrid** will display it according to its position in this array. Specify the column's [type](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#type) as *"buttons"* and declare the [buttons](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/buttons/) array. It should contain your custom buttons and values for the predefined buttons.
```

File: JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md [hunk 3]
```diff
@@ -1 +1,3 @@
-The **DataGrid** widget allows you to&nbsp;group column headers logically to&nbsp;implement a&nbsp;banded layout. In&nbsp;this demo, the **columns** array has a&nbsp;hierarchical structure, although such a&nbsp;layout can be&nbsp;built on&nbsp;a&nbsp;flat-structured array as&nbsp;well. Note that despite being banded, the nested columns remain interactive.
\ No newline at end of file
+To implement multi-level headers, transform the [columns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/) array into a hierarchical one as done in this demo. If you don't use the **columns** array, set the band and banded columns in the [ownerBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#ownerBand) and [isBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#isBand) options inside the customizeColumns function (see the example in the **isBand** description).
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md]: customizeColumns should be a link too.

File: JSDemos/menuMeta.json [hunk 4]
```diff
@@ -522,7 +522,7 @@
             "DemoType": "Web"
           },
           {
-            "Title": "Multi-Row Headers (Bands)",
+            "Title": "Multi-Level Headers (Bands)",
             "Name": "MultiRowHeadersBands",
             "DocUrl": "Guide/Widgets/DataGrid/Columns/Column_Types/Band_Columns/",
```
Comment (RomanTsukanov) [inline JSDemos/menuMeta.json]: Remove doc URLs for these descriptions.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md [hunk 5]
```diff
@@ -1 +1,3 @@
-The **DataGrid** widget allows you to&nbsp;group column headers logically to&nbsp;implement a&nbsp;banded layout. In&nbsp;this demo, the **columns** array has a&nbsp;hierarchical structure, although such a&nbsp;layout can be&nbsp;built on&nbsp;a&nbsp;flat-structured array as&nbsp;well. Note that despite being banded, the nested columns remain interactive.
\ No newline at end of file
+To implement multi-level headers, transform the [columns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/) array into a hierarchical one as done in this demo. If you don't use the **columns** array, set the band and banded columns in the [ownerBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#ownerBand) and [isBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#isBand) options inside the [customizeColumns ](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#customizeColumns)function (see the example in the **isBand** description).
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md]:
```suggestion
To implement multi-level headers, transform the [columns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/) array into a hierarchical one as done in this demo. If you don't use the **columns** array, set the band and banded columns in the [ownerBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#ownerBand) and [isBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#isBand) options inside the [customizeColumns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#customizeColumns) function (see the example in the **isBand** description).  
```

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/DataGrid/ColumnCustomization/description.md [hunk 6]
```diff
@@ -1 +1,25 @@
...
+Invoke a context menu in a column's header and specify whether to fix the column to the left or right. The fixed column remains visible when users scroll the view horizontally.
+
+* Widget option: **columnFixing**.[enabled](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columnFixing/#enabled)
+* Column option: **columns[]**.[fixed](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#fixed) - if enabled, fixes a column to the left; **columns[]**.[fixedPosition](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#fixedPosition) - specifies whether to fix a column to the left or right
+
+**Show and hide columns**
+Click the Column Chooser button to access hidden columns. Drag the column's header between the Column Chooser and the grid to change the column's visibility.
+
+* Widget option: **columnChooser**.[enabled]/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columnChooser/#enabled)
+* Column option: columns[].[visible](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#visible)
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/ColumnCustomization/description.md]:
```suggestion
* Column option: **columns[]**.[visible](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#visible)
```

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md [hunk 7]
```diff
@@ -1,3 +1,19 @@
...
 
-The customizations mentioned above are implemented in&nbsp;the <a href="/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/">editing</a> object for standard <b>Edit</b> and <b>Delete</b> buttons, and in&nbsp;the <a href="/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/buttons/">buttons</a> array for the custom <b>Clone</b> button.
\ No newline at end of file
+**Predefined buttons**           
+The available selection of predefined buttons depends on the selected edit mode, and the availability of the update and delete operations. To enable them, set the [allowUpdating](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowUpdating) and [allowDeleting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowDeleting) options to **true**. You can also set any of these options to a custom function that enables the corresponding operation and displays its button only for certain rows.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md]: Let's give a link to the edit mode option.

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md [hunk 8]
```diff
@@ -1,3 +1,19 @@
...
+
+This demo has the following configuration:
+
+* edit mode is set to *"row"*
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md]:
```suggestion
* Edit mode is set to *"row"*.
```

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md [hunk 9]
```diff
@@ -1,3 +1,19 @@
...
+
+* **allowUpdating** is enabled to display Edit buttons in all rows
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md]:
```suggestion
* **allowUpdating** is enabled to display Edit buttons in all rows.
```

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md [hunk 10]
```diff
@@ -1,3 +1,19 @@
...
+  
+* **allowDeleting** is set to a function that hides the Delete button based on a custom condition
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md]:
```suggestion
* **allowDeleting** is set to a function that hides the Delete button based on a custom condition.
```

File: JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md [hunk 11]
```diff
@@ -1 +1,9 @@
-The **DataGrid** widget allows you to&nbsp;group column headers logically to&nbsp;implement a&nbsp;banded layout. In&nbsp;this demo, the **columns** array has a&nbsp;hierarchical structure, although such a&nbsp;layout can be&nbsp;built on&nbsp;a&nbsp;flat-structured array as&nbsp;well. Note that despite being banded, the nested columns remain interactive.
\ No newline at end of file
+Our **DataGrid** widget allows you to group multiple columns under one header (band). This demo shows an example: the "Nominal GDP" and "Population" bands have banded columns.
+
+You can drag the band across the grid to reorder all its banded columns simultaneously. You can also move the band to the column chooser and hide the banded columns. Banded columns remain interactive.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md]:
```suggestion
You can drag one of the bands across the grid to reorder all its banded columns simultaneously. You can also move the band to the column chooser and hide the banded columns. Banded columns remain interactive.
```

File: JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md [hunk 12]
```diff
@@ -1 +1,9 @@
...
+
+To create the banded layout, do one of the following:
+
+* Assign a hierarchical structure to the columns option. The code in this demo is an example.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md]: Let's give a link to the columns option.

File: JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md [hunk 13]
```diff
@@ -1 +1,9 @@
...
+
+* Implement a [customizeColumns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#customizeColumns) function where you should specify the [ownerBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#ownerBand) and [isBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#isBand) column options. Refer to the isBand description for a code sample.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/MultiRowHeadersBands/description.md]:
```suggestion
* Implement a [customizeColumns](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#customizeColumns) function where you should specify the [ownerBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#ownerBand) and [isBand](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/#isBand) column options. Refer to the **isBand** description for a code sample.
```

File: JSDemos/menuMeta.json [hunk 14]
```diff
@@ -522,9 +522,8 @@
             "DemoType": "Web"
           },
           {
-            "Title": "Multi-Row Headers (Bands)",
+            "Title": "Multi-Level Headers (Bands)",
             "Name": "MultiRowHeadersBands",
-            "DocUrl": "Guide/Widgets/DataGrid/Columns/Column_Types/Band_Columns/",
```
Comment (RomanTsukanov) [inline JSDemos/menuMeta.json:527]: We better remove `DocUrl`'s from other demos from this section, too.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: JSDemos/menuMeta.json (hunk 14 above)
Comment (pavel-2020) [reply JSDemos/menuMeta.json:527]: Should "Name": "MultiRowHeadersBands", be changed to "Name": "MultiLevelHeadersBands"?

Comment (RomanTsukanov) [reply JSDemos/menuMeta.json:527]: No, "Name" points at a folder with the demo.

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md [hunk 15]
```diff
@@ -1,3 +1,19 @@
...
 
-The customizations mentioned above are implemented in&nbsp;the <a href="/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/">editing</a> object for standard <b>Edit</b> and <b>Delete</b> buttons, and in&nbsp;the <a href="/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/columns/buttons/">buttons</a> array for the custom <b>Clone</b> button.
\ No newline at end of file
+**Predefined buttons**           
+The available selection of predefined buttons depends on the selected [edit](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#mode) mode, and the availability of the update and delete operations. To enable them, set the [allowUpdating](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowUpdating) and [allowDeleting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowDeleting) options to **true**. You can also set any of these options to a custom function that enables the corresponding operation and displays its button only for certain rows.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/description.md]:
```suggestion
The available selection of predefined buttons depends on the selected [edit mode](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#mode), and the availability of the update and delete operations. To enable them, set the [allowUpdating](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowUpdating) and [allowDeleting](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/editing/#allowDeleting) options to **true**. You can also set any of these options to a custom function that enables the corresponding operation and displays its button only for certain rows.
```

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**